lib/Router.js

```diff
@@ -250,6 +250,20 @@ Router.prototype.handle = function(request, connection) {
         return request.error(404, "No handlers available for resource " + resourceId);
     }
 
+    if (!!~[Request.METHOD_PATCH, Request.METHOD_POST, Request.METHOD_PUT].indexOf(method)) {
+        var validate = jsen(_.extend({additionalProperties: false}, route.schema));
+        var resource = request.getResource();
+        if (resource) {
+            delete resource.$id;
+            delete resource.$expires;
+            var valid = validate(resource);
+            if (!valid) {
+                emit.call(this, 'api:error', request, null, timeStart);
+                return request.error(422, "Invalid schema for resource " + resourceId);
```

Comment:
The error message is incorrect. It should be something like "Provided resource did not validate with schema."